### ISSUE TITLE
Configuration system rework

### ISSUE THREAD

**Issue by unorthodox-paradox** [unorthodox-paradox/omsi-2_csb-ai-enhancements#87]
#### Problem statement ####

Currently the only means of configuring the vehicles that is officially supported is via plain .cti-variables. Users assign these via `[setvar]`, and the scripts simply operate directly based on their values (or absence thereof).

This approach, while simple, is becoming increasingly inadequate as the script base expands:
- As is the case with every uninitialized variable in OMSI, the default for .cti-vars is always zero. This is inconvenient because a) in some cases a different default value would make more sense; and worse yet, b) in others a zero value makes no sense at all, be it default or not. As a clumsy workaround for (a), and specifically in the case of boolean (zero vs. non-zero) variables / flags, we have been naming such variables with a negative prefix (e.g. `xxx_cti_do_not_do_xyz`), so as to convey an inversion of semantics, i.e., that the non-default value "takes away", rather than "adds", behavior. Needless to say, this strategy is bothersome to maintain and confusing to users. As for (b), there has been no consistent workaround -- such variables are still dealt with as individual exceptions, further hurting maintainability
- .cti-vars are effectively set in stone when assigned via `[setvar]`. OMSI does not merely perform the instructed assignments once, during initialization, but in perpetuity. Given that we rely on those variables directly, this has the important consequence that neither a) can we re-assign them with a "fall-back" default in case they are invalid, nor b) can the user re-assign them post-spawn (e.g. through custom triggers or some sort of configuration utility (display, "magic book", etc.) to be provided by us in a future version). The fact that we can support neither proper validation nor dynamic re-assignment of configuration poses a significant "roadblock" to future development.
- The fact that there is no .cti-file for the default (unnamed) color-scheme, effectively precludes its configuration / customization.

#### Course of action ####

To overcome the aforementioned limitations, a more comprehensive configuration system is obviously needed. The new system should support configuration at various levels (global configurable and fall-back defaults; color-scheme-specific overrides; dynamic in-game overrides), without compromising maintainability and extensibility.

Requirements draft:
- At the lowest level, L0, there ought to be hard-coded, sensible defaults for all configuration variables. In the absence of overriding, valid configuration at any higher level (see below), these values are to be considered by scripts as being in effect.
- At L1, there ought to be user-configurable defaults in the form of constants, to be applied to all vehicles regardless of color-scheme in use. When valid, these shall override the configuration at L0.
- At L2, there ought to be user-configurable, color-scheme-specific overrides of the respective constants at L1, or hard-coded fall-back defaults at L0. These are to be processed only once, upon the vehicle's initial spawning, so as to leave room for further dynamic in-game re-configuration. Furthermore, there ought to be constants / flags for both a) selectively disabling individual L2 overrides, or b) globally disabling them, in favor of L0/L1 configuration. These flags should not affect acknowledgement of L3 overrides (see below).
- At L3, there ought to be dynamically-configurable overrides of the respective lower-level variables (L2), constants (L1), or hard-coded defaults (L0), for use with custom triggers or our own configuration utility yet to be implemented. These shall be continuously observed for changes, commencing right after L2-processing has completed.

Implementation considerations:
- Scripts must have their own internal copy of effective configuration variables, and operate on those, as opposed to L2/L3 variables that the user is supposed to (be able to) influence. This isn't so much to prevent users from intentionally breaking scripts through invalid assignments (which they after all always find a creative way to accomplish if they so please), but to improve robustness and (even if only so slightly) performance.
- This is a good opportunity for most scripts that are configurable at present, or will likely have to become configurable in the future, to adopt a consistent initialization pattern:
    1. During `{init}`:
        - If the vehicle has been spawned for the very first time, perform any one-time initialization tasks that are independent of external context (e.g., weather) that only gets communicated to scripts post-init. Furthermore, apply L1 overrides of L0 configuration, if valid, and construct an initial internal valid configuration set.
        - Also carry out any other initialization tasks that actually ought to be repeated during each spawn; e.g., resetting `GetTime`-based timers.
   1. During `{frame}`:
        - During the first execution, do not execute any logic dependent on configuration or other external context, which may not have been initialized by OMSI yet.
        - During the second execution, if the vehicle has been spawned for the very first time, apply L2 overrides of L0/L1 configuration, if valid and not disabled. Also perform any further initialization previously deferred due to dependence on external context.
       - During the second and any further execution, apply L3 overrides of L0/L1/L2 configuration, if valid.

Non-goals (for now):
- Provide an additional, "special" configuration level, L2*, modeled as constants, for use with the default (unnamed) color-scheme, given that it lacks a .cti-file and thus also the ability of being configured via L2 / `[setvar]`s.
- Implement a friendly in-game configuration utility (display, "magic book", etc.).

#### Task-list ####

Scripts (and corresponding var-lists and const-files) to be revised:
- [x] `lights.osc`
- [x] `cockpit.osc`
- [x] `door.osc`
- [x] `wiper.osc`
- [x] `elec.osc`*
- [x] `ZF-Eco.osc`* (antrieb)
- [x] `engine.osc`*
- [x] `bremse.osc`*
- [x] `IBIS-2.osc`*
- [x] `ticketprinter.osc`*
- [x] `uchill.osc`
- [x] `VDV-old.osc`*
- [x] `visual.osc`

(items marked '*' are not configurable at present, and shall merely get equipped with the revised initialization flow for the time being)

- [x] Obviously, all other scripts, along with model- and sound-configs, should be checked for occurrences of access of configuration variables other than their own, and revised if need be, such that they only read _effective_ configuration and not configuration update requests at any level.

- [x] Lastly, readme/A.2 should be updated to reflect the changes.

**Comment by unorthodox-paradox** [comment]
Unfortunately it looks like OMSI does not consistently apply `[setvar]`s starting from the first post-init frame (it does so _usually_... but not always). As a consequence, rather than being treated as a one-time deferred initialization task, L2 override processing now occurs alongside L3 override processing, once per frame. Sub-optimal as this workaround may seem, it's also the sanest.
